bin2genga.c

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copied from bin2ascii.c
+ * Convert TIPSY format into GENGA format, with
+ * << x y z m vx vy vz r >>
+ */
+/*
+ */
+
+#include <stdio.h>
+#include <malloc.h>
+#include "tipsydefs.h"
+
+int
+main()
+{
+    struct dark_particle *dark_particles, *dp, *lastdp;
+    struct dump header;
+    
+    while(1) 
+      {
+	if(fread((char *)&header,sizeof(header),1,stdin) == 0)
+	  break;
+	if(header.ndark != 0) {
```

Comment:
It looks like you assume that nSph == 0 and nStar == 0.  You should check and assert on that.